builtin/mainmenu/dlg_create_world.lua

```diff
@@ -150,11 +379,60 @@ local function create_world_buttonhandler(this, fields)
 		return true
 	end
 
+	for k,v in pairs(fields) do
+		local split = string.split(k, "_", nil, 3)
+		if split and split[1] == "flag" then
+			local setting
+			if split[2] == "mg" then
+				setting = "mg_flags"
+			else
+				setting = split[2].."_spflags"
+			end
+			-- We replaced the underscore of flag names with a dash.
+			local flag = string.gsub(split[3], "-", "_")
+			local ftable = flags_to_table(core.settings:get(setting))
+			if v == "true" then
+				ftable[flag] = true
+			else
+				ftable[flag] = false
+			end
+			local flags = table_to_flags(ftable)
+			core.settings:set(setting, flags)
+			return true
+		end
+	end
+
 	if fields["world_create_cancel"] then
 		this:delete()
 		return true
 	end
 
+	if fields["mgv6_biomes"] then
```

Comment:
No separate treatment for v6 please.

Reply:
Docs already explain how v6 flags are interpreted:

#    The 'snowbiomes' flag enables the new 5 biome system.
#    When the 'snowbiomes' flag is enabled jungles are automatically enabled and
#    the 'jungles' flag is ignored.

Reply:
This generic GUI is to make it intuitive for any user so they get frustrated when one combination does not work. 
Special treatment is totally fine to me.

However, _please_ combine the dropdown entries and the setting overrides. Example:
```Lua
local mgv6_biomes = {
	{
		fgettext("Temperate, Desert, Jungle, Tundra, Taiga"),
		{jungles = true, snowbiomes = true}
	},
	{
		fgettext("Temperate, Desert, Jungle"),
		{jungles = true, snowbiomes = false}
	},
	{
		fgettext("Temperate, Desert"),
		{jungles = false, snowbiomes = false}
	}
}
```
1) `dropdown` return values can be indexed directly
2) No `string.split` hack
3) Setting overrides are specified directly with the description
4) Allows simple iterating